core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -872,6 +917,427 @@ export class ExplorationEditor extends BaseUser {
+        );
+      }
+
+      showMessage(
+        `Revision ${version} has the property ${property} with value: ${value}`
+      );
+    }
+  }
+
+  /**
+   * Checks if the revisions are ordered by a specific property.
+   * @param {string} property - The property to check the order by.
+   */
+  async expectRevisionsToBeOrderedBy(
+    property: string,
+    order: 'asc' | 'desc'
+  ): Promise<void> {
+    if (!property) {
+      throw new Error(`Invalid property: ${property}`);
+    }
+
+    const propertyToSelector: {[key: string]: string} = {
+      'Version No.': revisionVersionNoSelector,
+      Notes: revisionNoteSelector,
+      User: revisionUsernameSelector,
+      Date: revisionDateSelector,
+    };
+
+    let elementHandles = await this.page.$$(historyListItem);
+    let revisions: {[key: string]: string | Date | number}[] =
+      await Promise.all(
+        elementHandles.map(async handle => {
+          let valueHandle = await handle.$(propertyToSelector[property]);
+          if (!valueHandle) {
+            throw new Error(`Failed to retrieve value handle for ${property}`);
+          }
+          let value = await valueHandle.getProperty('textContent');
+          return {
+            [property]: (await value.jsonValue()) as string | Date | number,
+          };
+        })
+      );
+
+    for (let i = 0; i < revisions.length - 1; i++) {
+      let value1 = revisions[i][property];
+      let value2 = revisions[i + 1][property];
+
+      if (value1 === undefined || value2 === undefined) {
+        throw new Error(
+          `Undefined value for ${property} in one or more revisions`
+        );
+      }
+
+      if (typeof value1 === 'string' && !isNaN(Date.parse(value1))) {
+        value1 = new Date(value1);
+        value2 = new Date(value2);
+      }
+
+      if (value1 instanceof Date && value2 instanceof Date) {
+        value1 = value1.getTime();
+        value2 = value2.getTime();
+      }
+
+      if (value1 === value2) {
+        throw new Error(`Two revisions have the same ${property}: ${value1}`);
+      } else if (order === 'asc' ? value1 > value2 : value1 < value2) {
+        throw new Error(
+          `Revisions are not sorted by ${property} in ${order} order`
+        );
+      }
+    }
+
+    showMessage(`Revisions are sorted by ${property} in ${order} order`);
+  }
+
+  /**
+   * Filters the revisions by a specific user.
+   * @param {string} userName - The name of the user to filter the revisions by.
+
+   */
+  async filterRevisionsByUser(userName: string): Promise<void> {
+    if (!userName) {
+      throw new Error('Username cannot be empty');
+    }
+
+    await this.type(userNameEdit, userName);
+    // The unicode '\u000d' is used to simulate the Enter key press.
+    await this.type(userNameEdit, '\u000d');
+  }
+
+  /**
+   * Checks if the number of revisions matches the expected number.
+   * @param {number} expectedNumber - The expected number of revisions.
```

Comment:
I think this should be renamed to `expectedRevisionCount` as it better describes what the parameter represents as opposed to how its represented in the programming language (as the type system takes care of that).

Reply:
sure.